v2/yarpcroundrobin/list_test.go

```diff
@@ -20,887 +20,490 @@
+				yarpctest.UpdateAction{AddedPeerIDs: []string{"1"}, ExpectedErr: yarpcpeer.ErrInvalidPeerType{}},
+			},
+		},
+		{
+			msg: "update retain multiple errors",
+			retainedAvailablePeerIDs: []string{"2"},
+			errRetainedPeerIDs:       []string{"1", "3"},
+			retainErr:                yarpcpeer.ErrInvalidPeerType{},
+			peerListActions: []yarpctest.PeerListAction{
+				yarpctest.UpdateAction{
+					AddedPeerIDs: []string{"1", "2", "3"},
+					ExpectedErr:  multierr.Combine(yarpcpeer.ErrInvalidPeerType{}, yarpcpeer.ErrInvalidPeerType{}),
+				},
+			},
+			expectedAvailablePeers: []string{"2"},
+		},
+		{
+			msg: "add retain error",
+			retainedAvailablePeerIDs: []string{"1", "2"},
+			expectedAvailablePeers:   []string{"1", "2"},
+			errRetainedPeerIDs:       []string{"3"},
+			retainErr:                yarpcpeer.ErrInvalidPeerType{},
+			peerListActions: []yarpctest.PeerListAction{
+				yarpctest.UpdateAction{AddedPeerIDs: []string{"1", "2"}},
+				yarpctest.UpdateAction{
+					AddedPeerIDs: []string{"3"},
+					ExpectedErr:  yarpcpeer.ErrInvalidPeerType{},
+				},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+				yarpctest.ChooseAction{ExpectedPeer: "2"},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+			},
+		},
+		{
+			msg: "add duplicate peer",
+			retainedAvailablePeerIDs: []string{"1", "2"},
+			expectedAvailablePeers:   []string{"1", "2"},
+			peerListActions: []yarpctest.PeerListAction{
+				yarpctest.UpdateAction{AddedPeerIDs: []string{"1", "2"}},
+				yarpctest.UpdateAction{
+					AddedPeerIDs: []string{"2"},
+					ExpectedErr:  yarpcpeer.ErrPeerAddAlreadyInList("2"),
+				},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+				yarpctest.ChooseAction{ExpectedPeer: "2"},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+			},
+		},
+		{
+			msg: "remove peer not in list",
+			retainedAvailablePeerIDs: []string{"1", "2"},
+			expectedAvailablePeers:   []string{"1", "2"},
+			peerListActions: []yarpctest.PeerListAction{
+				yarpctest.UpdateAction{AddedPeerIDs: []string{"1", "2"}},
+				yarpctest.UpdateAction{
+					RemovedPeerIDs: []string{"3"},
+					ExpectedErr:    yarpcpeer.ErrPeerRemoveNotInList("3"),
+				},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+				yarpctest.ChooseAction{ExpectedPeer: "2"},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+			},
+		},
+		{
+			msg: "remove release error",
+			retainedAvailablePeerIDs: []string{"1", "2"},
+			errReleasedPeerIDs:       []string{"2"},
+			releaseErr:               yarpcpeer.ErrDialerHasNoReferenceToPeer{},
+			expectedAvailablePeers:   []string{"1"},
+			peerListActions: []yarpctest.PeerListAction{
+				yarpctest.UpdateAction{AddedPeerIDs: []string{"1", "2"}},
+				yarpctest.UpdateAction{
+					RemovedPeerIDs: []string{"2"},
+					ExpectedErr:    yarpcpeer.ErrDialerHasNoReferenceToPeer{},
+				},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+				yarpctest.ChooseAction{ExpectedPeer: "1"},
+			},
+		},
+		// Flaky in CI
```

Comment:
I realize this is a relic from the past, but we should figure out this flakiness. Maybe update the `CHANGES` file if it's not already there?